libelement/src/instruction_tree/evaluator.cpp

```diff
@@ -22,6 +22,14 @@ static void add_to_cache(instruction_cache_value* cache_value, element_value val
 static element_result do_evaluate(element_evaluator_ctx& context, const element::instruction_const_shared_ptr& expr,
                                   instruction_cache* cache, element_value* outputs, size_t outputs_count, size_t& outputs_written)
 {
+    // Don't cache constants, faster to grab the value
```

Comment:
We have still allocated memory in the cache for constants though.

Is it worth excluding constants (and inputs, and multi-valued things) when initialising the cache? This would reduce the number of comparisons needed every time anything is read in the cache.

I'd be interested to see some numbers, shouldn't be too big a change to do either...

👀

Reply:
I'll take a look :) I wouldn't expect much of an improvement though given how map lookups work, but let's see 😁

Reply:
donesies, it is a bit faster actually :) thanks